Make handling of amount safer (do not use floats)

ITspirit opened the pull request:
With money math the behaviour of different "float-styles" is safely omitted. Works with Heidelpay - no need to send a "float".

ITspirit committed:
Make handling of amount safer (do not use floats)

ITspirit committed:
better version - without the need of an additional extension (but supported it they are present like gmp)

albertmueller commented on src/CoreShop/Payum/HeidelpayBundle/Extension/PopulateHeidelpayExtension.php:
```diff
@@ -99,24 +101,32 @@ public function onPostExecute(Context $context)
-        $result['basket']['amount'] = round($result['basket']['amount'] / 100, 2);
-
-        $request->setResult((array)$result);
+        $result['basket']['amount'] = $this->calcAmount($result['basket']['amount']);
```
i think $payment->getCurrencyCode() can be used here as second parameter, otherwise it's fixed to EUR

ITspirit replied:
Makes in this case not a really difference (we just format the amount value), but you are right. I've added this :-)

ITspirit committed:
- add dynamic paymentCurrencyCode
- add and throw HeidelpayBundleException in case of missing customer or invoice address

ITspirit commented:
Can you make a release (1.0.2 or 1.1.0), so it´s usable in the projects? Would be great! Thx

dpfaffenbauer commented:
Honestly: I would have not merged it that fast. I really don't like the moneyphp dependency, and I would have solved it differently. Not sure how exactly, but there would be a way.

So, since this is now merged, we should leave it as is, and release 1.1.0, can you make that @albertmueller `

ITspirit commented:
@dpfaffenbauer I´ve no problem with a dependency, especially with a perfect one like money. 

Easy way would be to convert the int amount to a string and then split off the last two numbers and concat it. But, to be honest, that´s not a good practice.

Maybe we can discuss this for "2.0", because i´am working to upgrade this and the other heidelpay bundle to the new heidelpay-sdk.

albertmueller commented:
@dpfaffenbauer i know, i also thought of another solution, in detail i found moneyphp not so bad. we're able to change it in the future, if we have a better solution.

dpfaffenbauer commented:
I am just thinking across the CoreShop ecosystem, and having moneyphp in one payment bundle isn't the best solution, other payment bundles might come across a similar issue. But yeah, lets see what happens and leave it as it is now.

albertmueller commented:
of course, u're right at this point
